### ISSUE TITLE
Add2App [Option B] product can't be uploaded to AppStoreConnect

### ISSUE THREAD

**Issue by nailgilaziev** [flutter/flutter#47930]
After step by step following the instruction [Integrate a Flutter module into your iOS project](https://flutter.dev/docs/development/add-to-app/ios/project-setup#option-b---embed-frameworks-in-xcode) app is successfully building, running and archiving but in the process of uploading to AppStoreConnect error occurs with message:
**Code signing "Flutter.framework" failed.**
```
[✓] Flutter (Channel stable, v1.12.13+hotfix.5, on Mac OS X 10.15.2 19C57, locale en-RU)
    • Flutter version 1.12.13+hotfix.5 at /Users/n/Tools/flutter
    • Framework revision 27321ebbad (3 weeks ago), 2019-12-10 18:15:01 -0800
    • Engine revision 2994f7e1e6
    • Dart version 2.7.0

[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at /Users/n/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.3, Build version 11C29
    • CocoaPods version 1.8.4

[✓] Android Studio (version 3.5)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 42.1.1
    • Dart plugin version 191.8593
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)

[✓] Connected device (1 available)
    • iPhone (Grado) • e9e4908a31af78e6cfd1ff8c7164b51aee8f6459 • ios • iOS 13.3
```

Steps to reproduce:

```
cd ~
mkdir pubios
cd pubios
flutter create --template module my_flutter
```
open xcode and create new project - Single View App (Storyboard based)
I call it `host_ios`

so file structure currently is next
```
pubios/
├─my_flutter
└─host_ios
```

generating flutter app frameworks (note: directly to host_ios folder)
```
cd my_flutter
flutter build ios-framework --output=../host_ios/embed/
```
Go to xcode and DO steps described in documentation [Option B - Embed frameworks in Xcode](https://flutter.dev/docs/development/add-to-app/ios/project-setup#option-b---embed-frameworks-in-xcode)

- Go to settings -> _General_ tab 
- Press app/plus button in section _Frameworks, Libraries, and Embedded Content_
- Press buttun _Add Other..._ -> _Add Files_
- Navigate to folder `../host_ios/embed/Release/` where frameworks saved
- Select both **App.framework** and **Flutter.Framework**
- open _Build Settings_ tab and search for Framework Search Paths (FRAMEWORK_SEARCH_PATHS)
- change hardcoded value from `$(PROJECT_DIR)/embed/Release`to `$(PROJECT_DIR)/embed/$(CONFIGURATION)` 
- open file `../host_ios/host_ios.xcodeproj/project.pbxproj` in some text editing tool
- replace row `path = embed/Release/App.framework;`to `path = "embed/$(CONFIGURATION)/App.framework";`
- do same for Flutter.Framework 
- save, exit ang go to xcode (your frameworks icons now has some white color in settings)

Open ViewController.swift and replace content with this code
```
import UIKit
import Flutter

class ViewController: UIViewController {
    
    override func viewDidLoad() {
        super.viewDidLoad()
        let button = UIButton(frame: CGRect(x: 140, y: 200, width: 100, height: 50))
        button.setTitle("run flutter", for: .normal)
        button.addTarget(self, action: #selector(buttonAction), for: .touchUpInside)
        self.view.addSubview(button)
    }
    
    lazy var flutterEngine = FlutterEngine(name: "flutter engine for reports")
    var flutterVC:FlutterViewController? = nil
    
    @objc func buttonAction(sender: UIButton!) {
        if flutterVC == nil{
            // Runs the default Dart entrypoint with a default Flutter route.
            // flutterEngine field initialized lazily
            flutterEngine.run();
            flutterVC = FlutterViewController(engine: flutterEngine, nibName: nil, bundle: nil)
        }
        present(flutterVC!, animated: true, completion: nil)
    }
}
``` 
Run and you should see the integrated to the host ios app default flutter counter app presented after button tap.

Now, lets try to upload this app to appstore connect!
Follow the documentation tutorial [Build and release an iOS app](https://flutter.dev/docs/deployment/ios)

- I change BundleIdentivier, Version, Build number in _General_ tab
- And change Signing section in _Signing&Capabilities_ tab to appropriate values (Distribution provisionProfile and Distribution certificate)

Next, archive and upload product to appStore connect!

Code signing "Flutter.framework" failed.
![err](https://user-images.githubusercontent.com/2749021/71549773-204dda00-29d4-11ea-8494-e29b56d60b16.png)

Ps
My code signing entitlements is correct. Because if I remove added frameworks I can successfully sign and upload the app to appstore connect. 

In Distribution logs I cant find anything that describe error more detailed. All files attached
IDEDistribution.critical.log - Empty file
[IDEDistribution.standard.log](https://github.com/flutter/flutter/files/4007501/IDEDistribution.standard.log)
[IDEDistribution.verbose.log](https://github.com/flutter/flutter/files/4007502/IDEDistribution.verbose.log)
[IDEDistributionPipeline.log](https://github.com/flutter/flutter/files/4007503/IDEDistributionPipeline.log)
IDEDistributionProvisioning.log - empty file
[IDEITunesSoftwareService.log](https://github.com/flutter/flutter/files/4007504/IDEITunesSoftwareService.log)

```
.../host_ios/embed/Release    ls -l
drwxr-xr-x  5 n  staff  160 Dec 29 00:00 App.framework
drwxr-xr-x  8 n  staff  256 Dec 28 23:59 Flutter.framework
```

Default new Xcode project comes with Bitcode Enabled true

**Comment by nailgilaziev** [comment]
I try to check same on master channel 
```
flutter channel master
flutter upgrade
```
_it takes so long time..._
```
Flutter 1.13.6-pre.39 • channel master • https://github.com/flutter/flutter.git
Framework • revision 841924e161 (3 hours ago) • 2019-12-28 14:50:12 -0500
Engine • revision bdc9708d23
Tools • Dart 2.8.0 (build 2.8.0-dev.0.0 c547f5d933)
```
regenerate flutter app frameworks
```
cd ~/pubios/my_flutter
flutter build ios-framework --output=../host_ios/embed/
```
Open Xcode -> Product -> Clean build folder -> Archive
Distribute App -> Upload to AppStore
And...
Same issue:
**Code signing "Flutter.framework" failed.**

**Comment by nailgilaziev** [comment]
If I try to set bitcode enabled to false
and repeate process - during upload I go a step further and see this:

![app](https://user-images.githubusercontent.com/2749021/71550429-fe0e8900-29e0-11ea-9473-8d370133ccca.png)

Architectures: Intel 64 - why??

![flutter](https://user-images.githubusercontent.com/2749021/71550431-049d0080-29e1-11ea-8952-b863e8489507.png)
![errors](https://user-images.githubusercontent.com/2749021/71550433-0a92e180-29e1-11ea-9352-8d8e4c87eb11.png)

I remind, that frameworks produced by command
`flutter build ios-framework --output=../host_ios/embed/`

And I still in master channel (stable channel behave the same if I remember correctly)

**Comment by jmagman** [comment]
x86_64 is for the simulator.  Hmm, that arch isn't getting thinned...

Does it work if you embed the Flutter XCFramework instead since you're on >Xcode 11?  Same instructions if you want to use the `$CONFIGURATION` path except the extension is `.xcframework` instead of `.framework`.
```
flutter build ios-framework --output=../host_ios/embed/ --xcframework --no-universal
```
By the way, in stable there's [an issue](https://github.com/flutter/flutter/issues/46993#issuecomment-566713673) with using relative paths in `--output` so use absolute paths if you swap back.  It's fixed in master and dev channels.

**Comment by jmagman** [comment]
Also note it's [App.framework](https://github.com/flutter/flutter/issues/44492), not App.xcframework.

**Comment by nailgilaziev** [comment]
I upgraded to this: _(before merge if I understand correctly)_
`flutter upgrade`
`Flutter (Channel master, v1.13.6-pre.48, on Mac OS X 10.15.2 19C57, locale en-RU`
build xcframeworks -> applying them and uploading now is successfull! Thank you!

But If I add some plugin dependencies to this module and try to upload it again:
(on build/archiving step I get  this error:)
> bitcode bundle could not be generated because '*.framework/*' was built without full bitcode. All frameworks and dylibs for bitcode must be generated from Xcode Archive or Install build file '*.framework/*' for architecture arm64

@jmagman , I opened another issue and post details here #48092
Thank you again!
